### PR TITLE
'code-push link' command

### PR DESCRIPTION
**Pull request by silhouettes**
This allows you to link accounts on the CodePush server.

### CODE REVIEW

File: cli/script/command-parser.ts [hunk 1]
```diff
@@ -256,6 +256,16 @@ var argv = yargs.usage(USAGE_PREFIX + " <command>")
 
         addCommonConfiguration(yargs);
     })
+    .command("link", "Link a new authentication provider to an account with the CodePush server", (yargs: yargs.Argv) => {
```
Comment (lostintangent) [inline cli/script/command-parser.ts]: "Link an additional authentication provider (e.g. GitHub) to an existing CodePush account"?

File: cli/script/command-executor.ts [hunk 2]
```diff
@@ -557,6 +571,7 @@ function login(command: cli.ILoginCommand): Promise<void> {
 
 function loginWithExternalAuthentication(action: string, serverUrl?: string): Promise<void> {
     initiateExternalAuthenticationAsync(action, serverUrl);
+    log("");    // Insert newline
```
Comment (shishirx34) [inline cli/script/command-executor.ts]: why?

**Comment by lostintangent** [comment]
LGTM!

**Comment by shishirx34** [comment]
LGTM

File: cli/script/command-executor.ts (hunk 2 above)
Comment (silhouettes) [inline cli/script/command-executor.ts]: Because I removed '/r/n' from `initiateExternalAuthentication...()`, which is used by `link`

**Pull request merged**